File: python/serve_compare_transfer_b0b2d1.py
#!/usr/bin/env python3
# serve_compare_transfer.sh — portiert nach python
# Quelle: shell, OpenClaw@gateway2:scripts/serve_compare_transfer.sh
# Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

import os
import sys
import subprocess
import argparse
from pathlib import Path
from http.server import HTTPServer, SimpleHTTPRequestHandler
import socket

# Konfiguration
COMPARE_DIR = "/home/openclaw/.openclaw/workspace/vscode/compare"
TRANSFER_DIR = "/home/openclaw/.openclaw/workspace/vscode/compare/transfer"
HOST_IP = "89.58.15.220"
PORT = 80
SELF_PATH = os.path.realpath(__file__)

def get_files():
    """Finde alle Dateien im Vergleichsverzeichnis außer dem Skript selbst"""
    compare_path = Path(COMPARE_DIR)
    self_path = Path(SELF_PATH)
    
    files = []
    for item in compare_path.iterdir():
        if item.is_file() and item != self_path:
            files.append(item)
    
    # Sortiere alphabetisch
    files.sort()
    return files

def main():
    files = get_files()
    
    if not files:
        print(f"Keine Dateien in {COMPARE_DIR} gefunden.")
        sys.exit(1)
    
    print()
    print(f"Bereitgestellte Dateien aus {COMPARE_DIR}:")
    for src in files:
        print(f"- {src.name}")
    
    print()
    print(f"Copy/Paste auf anderem Gateway (Download nach {TRANSFER_DIR}):")
    for src in files:
        file_name = src.name
        print(f"curl -fL --retry 3 --connect-timeout 10 -o {TRANSFER_DIR}/{file_name} http://{HOST_IP}:{PORT}/{file_name}")
    
    print()
    print(f"Server auf Port {PORT} aktiv. Beenden mit STRG+C.")
    print()
    
    # Wechsle ins Vergleichsverzeichnis und starte HTTP-Server
    os.chdir(COMPARE_DIR)
    
    # Starte HTTP-Server auf allen Interfaces
    httpd = HTTPServer(('0.0.0.0', PORT), SimpleHTTPRequestHandler)
    httpd.serve_forever()

if __name__ == "__main__":
    main()

File: javascript/serve_compare_transfer_b0b2d1.js
#!/usr/bin/env node
// serve_compare_transfer.sh — portiert nach javascript
// Quelle: shell, OpenClaw@gateway2:scripts/serve_compare_transfer.sh
// Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

const fs = require('fs');
const path = require('path');
const http = require('http');

const COMPARE_DIR = "/home/openclaw/.openclaw/workspace/vscode/compare";
const TRANSFER_DIR = "/home/openclaw/.openclaw/workspace/vscode/compare/transfer";
const HOST_IP = "89.58.15.220";
const PORT = "80";
const SELF_PATH = __filename;

// Funktion zum rekursiven Lesen von Dateien (ähnlich find)
function readFilesSync(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  
  list.forEach(file => {
    file = path.resolve(dir, file);
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) {
      // Überspringe Unterverzeichnisse
    } else {
      if (file !== SELF_PATH) {
        results.push(file);
      }
    }
  });
  
  return results.sort();
}

const files = readFilesSync(COMPARE_DIR);

if (files.length === 0) {
  console.log(`Keine Dateien in ${COMPARE_DIR} gefunden.`);
  process.exit(1);
}

console.log();
console.log(`Bereitgestellte Dateien aus ${COMPARE_DIR}:`);
files.forEach(src => {
  console.log(`- ${path.basename(src)}`);
});

console.log();
console.log(`Copy/Paste auf anderem Gateway (Download nach ${TRANSFER_DIR}):`);
files.forEach(src => {
  const file = path.basename(src);
  console.log(`curl -fL --retry 3 --connect-timeout 10 -o ${TRANSFER_DIR}/${file} http://${HOST_IP}:${PORT}/${file}`);
});

console.log();
console.log(`Server auf Port ${PORT} aktiv. Beenden mit STRG+C.`);
console.log();

// HTTP-Server erstellen
const server = http.createServer((req, res) => {
  const filePath = path.join(COMPARE_DIR, req.url === '/' ? 'index.html' : req.url);
  
  fs.access(filePath, fs.constants.F_OK, (err) => {
    if (err) {
      res.writeHead(404, { 'Content-Type': 'text/plain' });
      res.end('File not found');
      return;
    }
    
    const stat = fs.statSync(filePath);
    if (stat.isDirectory()) {
      res.writeHead(403, { 'Content-Type': 'text/plain' });
      res.end('Directory access forbidden');
      return;
    }
    
    res.writeHead(200, {
      'Content-Type': 'application/octet-stream',
      'Content-Length': stat.size
    });
    
    const readStream = fs.createReadStream(filePath);
    readStream.pipe(res);
  });
});

server.listen(PORT, '0.0.0.0', () => {
  console.log(`Server läuft auf http://0.0.0.0:${PORT}`);
});
